aicsimageio/aics_image.py

```diff
@@ -55,10 +57,40 @@ def determine_reader(image: types.ImageLike, **kwargs: Any) -> ReaderType:
         # numpy, dask, etc.
         if isinstance(image, (str, Path)):
             path = str(image)
+
+            # Check for extension in FORMAT_IMPLEMENTATIONS
```

Comment:
The if condition here brings up the case:  what if I have a s3 object with just a guid or lacking a file extension, but I want to tell AICSImage that I know the file type?  Is a kwarg in order for e.g. `reader_class="aicsimageio.readers.czi_reader.CziReader"` or something like that?

Reply:
Ermmmmm this check isn't trying the reader. That is done above. It is trying to write a naive guess at a better documented error.

Reply:
All this is doing is determining what install command to place in the returned error. The actual logic of checking if a file is supported is above where we test the bytes of the file, not the file name.

So in your exact example where they are using a guid or something. This error wouldn't occur at all, it would simply route to the next error that says:

```python
f"File extension suggests format: '{format_ext}'. "
f"Install extra format dependency with: "
f"`pip install aicsimageio[{format_ext}]`. "
f"See all known format extensions and their extra install "
f"name with `aicsimageio.formats.FORMAT_IMPLEMENTATIONS`."
```

Reply:
Thanks -- the collapsed code fooled me here. I'm good now.  The loop over all SUPPORTED_READERS is bad for scaling but something to think about later.